src/plugin/plugin.ts

```diff
@@ -83,6 +84,12 @@ export const createPlugin = <
             ...node.attrs,
             [elementSelectedNodeAttr]: isCurrentlySelected,
           });
+
+          if (selection instanceof NodeSelection) {
```

Comment:
This will happen for every descendant node – IIUC we just want it to happen once, if any element node has been modified.